engine/src/main/java/com/datatorrent/stram/cli/ApexCli.java

```
@@ -2178,6 +2178,15 @@ public void execute(String[] args, ConsoleReader reader) throws Exception
 
   }
 
+  private ApplicationReport findApplicationReportFromAppNameOrId(String appNameOrId)
+  {
+    ApplicationReport app = getApplication(appNameOrId);
+    if (app == null) {
+      app = getApplicationByName(appNameOrId);
+    }
+    return app;
```

All the callers throw the identical exception when the returned app is null. Can we just throw that exception here and the caller can always assume app is non-null?

---

The behavior seems to be inconsistent between different CL tools. I just checked  `docker start valid_container_name invalid_container_name`: this would start valid_container but not invalid_container. Then again when using `apt-get install valid_package_name invalid_package_name` it just fails completely, installing neither of those packages.

---

Making `findApplicationReportFromAppNameOrId()` throw a CLIException could 
- on one hand clean up the code so we don't need to rewrite it every time
- on the other hand it would make it inconsistent with getApplication and getApplicationByName, where no exception is thrown. 

Also if we leave the exception out of the method, we would still have the possibility of implementing the logic for "fail all vs. fail only for invalid appName" in the respective commands, which we do not have if we move it into the method

---

Ok

---

IMO, it should be the best effort attempt. If application id or name is not found, the error needs to be reported, but it is necessary to continue to the next id/name. At the end, command should report all applications that were shutdown.